### PR TITLE
Figure M2-Reconstructing Missingness

### PR DESCRIPTION
**Pull request by mcreixell**

### CODE REVIEW

**Review by aarmey — CHANGES_REQUESTED** [review]
You were overriding the `data` variable causing problems. I made a slightly simpler way of passing the data in by using `EM_clustering`. Look at the test.

I haven't tried your missingness code, but figure this will help in getting that setup.

**Comment by mcreixell** [comment]
This is what it looks like. Should we get rid of outliers?

Screen Shot 2020-10-21 at 11 19 39 AM

**Comment by aarmey** [comment]
Is the data already log-scaled here? A plot like this first makes me think that the data might be log-distributed...

**Comment by aarmey** [comment]
I'd be hesitant to remove those points solely based on their poor prediction.

**Comment by mcreixell** [comment]
It shouldn't be log-transformed. This is just the output of using `[mean_squared_error](https://scikit-learn.org/stable/modules/generated/sklearn.metrics.mean_squared_error.html) `. In the previous versions of this plot all errors were always between 0 and 1.

**Comment by aarmey** [comment]
I guess another way to ask what I’m wondering, is: are these points also the highest values, or are these distributed just like the rest of the data, but are predicted uniquely poorly?

**Comment by aarmey** [comment]
Either way I don’t think it’s fair to specifically eliminate them just because they are predicted poorly.

**Comment by mcreixell** [comment]
I found a couple errors but still don't explain these outliers. I will run this again and will add more information on the final matrix about each data point (e.g. peptide, tmt to be predicted, etc) to trace this back.

**Comment by mcreixell** [comment]
The outlier MSE come from outliers in the phosphorylation data:

Screen Shot 2020-10-22 at 11 39 01 PM

**Comment by aarmey** [comment]
Ok. I wouldn't remove them at this point, but maybe make note of them for later.

**Comment by mcreixell** [comment]
This is a regerssion plot for data, mix, and sequence, respectively. Maybe it'd be helpful to group data points by missingness % and average per fit? We'd end up with 3 data points per missingness group. @aarmey 

Screen Shot 2020-10-26 at 11 02 16 AM
Screen Shot 2020-10-26 at 11 02 20 AM
Screen Shot 2020-10-26 at 11 02 23 AM

**Comment by aarmey** [comment]
First, I think you want your y-axis log-scaled.

**Comment by mcreixell** [comment]
This is how it looks like log-transformed. Seems like mix works better than the other two extremes?  

Screen Shot 2020-10-26 at 1 30 42 PM

**Comment by mcreixell** [comment]
This is the same data but plotting the regression line of the baseline error instead of a lineplot. 

Screen Shot 2020-10-26 at 1 56 15 PM

**Comment by mcreixell** [comment]
Since I had the code pretty much set up already, I decided to explore the imputation error using different number of clusters. It looks like increasing the number of clusters helps the model. Beyond 21 and 24 clusters, respectively the model struggles to fit however. Maybe it would be worth refitting the cptac models with 20 clusters instead of 15? 

Screen Shot 2020-10-27 at 12 09 32 AM

**Comment by aarmey** [comment]
Sounds reasonable to me.

**Comment by mcreixell** [comment]
I think figure 2 is ready for now. I left the first spot blank to add a diagram explaining the reconstruction process. I'll make this today to use it on Thursday's meeting.

**Review by aarmey — APPROVED** [review]
So there's no change with different weights?

**Comment by mcreixell** [comment]
No change sadly...

**Comment by aarmey** [comment]
What's the difference from before? The cluster number?

**Comment by mcreixell** [comment]
No, the cluster number is the same. I think that the difference here is that I'm only loading one model, the one with the mixed weight, and refitting with different weights. Maybe if I run a new model from scratch with every different weight we see more differences.

**Comment by mcreixell** [comment]
I did this for the reconstruction error plot across different cluster numbers and it doesn't take that long.
